docs/testrunner-toolkit/configuration/espresso.md

```diff
@@ -111,6 +111,112 @@ __Example__:
     - "10.0"
 ```
 
+## `devices`
+
+__Description__: Field for defining devices request.
+
+A real device can be selected using two different manner:
+- Specific device selection, through device IDs
+- Filtered device selection, with several contraints such as `name`, `platformVersion`, etc...
+
+When an ID is specified, it is prioritary on other settings.
+
+__Type__: *[]object*
+
+__Example__:
+```yaml
+devices:
+  - name: "Google Pixel.*"
+    platformVersion: 8.1
+  - id: Google_Pixel_2_real_us
+```
+
+### `id`
+
+__Description__: Request a specific device by its id.
+
+__Type__:  *string*
+
+__Example__:
+
+```yaml
+        id: Google_Pixel_2_real_us
+```
+
+### `name`
+
+__Description__:  Request a device by its name.
+
+__Type__:  *string*
+
+__Example__:
+
+Using complete name:
+```yaml
+      - name: Google Pixel 4 XL
+```
+
+Using pattern matching:
+```yaml
+        name: Google Pixel.*
+```
+
+### `platformVersion`
+
+__Description__: Request a device with a specific platform version.
+
+__Type__:  *string*
+
+__Example__:
+
+```yaml
+        platformVersion: 8.1
+```
+
+### `options`
+
+__Description__:  Device request options.
+
+__Type__: *object*
+
+#### `carrierConnectivity`
+
+__Description__: Request a device that is connected to cellular network.
+
+__Type__:  *string*
+
+__Example__:
+
+```yaml
+       carrierConnectivity: true
+
+```
+
+#### `deviceType`
+
+__Description__:  Request a device from a specific kind. Values:  `ANY`, `TABLET`, `PHONE`.
+
+__Type__:  *string*
```

Comment:
Should this not be an enum, since the acceptable values are prescribed?

Reply:
Since it's a yaml file, enum is not really a thing. I don't know if we want `Type` to match the real yaml type or the expected implementation type. What's your point on this ?

Reply:
I guess it doesn't really matter. I just always assume that any element for which there is a prescribed list of valid values is an enum. :)

Reply:
I think that's better to keep it as it. We could discuss with the rest of the team during standup today !

Reply:
No need. I am fine with that decision. :)